OmgvaPOS/src/ItemManagement/Controller/ItemController.cs

```diff
@@ -161,5 +163,57 @@ public IActionResult DeleteItem(long id) {
                 return StatusCode((int)HttpStatusCode.InternalServerError, "Internal server error.");
             }
         }
+
+        [HttpGet("{id}/taxes")]
+        [Authorize(Roles = "Admin, Owner, Employee")]
+        [ProducesResponseType<List<TaxDto>>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)] //TODO: Should be thrown if item does not exist.
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public IActionResult GetItemTaxes(long id) {
+            if (!JwtTokenHandler.CanManageBusiness(HttpContext.Request.Headers.Authorization!, _itemService.GetItemNoException(id).BusinessId))
+                return Forbid();
+
+            try {
+                List<TaxDto> itemTaxes = _itemService.GetItemTaxes(id);
+                return Ok(itemTaxes);
+            }
+            catch (Exception ex) {
+                _logger.LogError(ex, "An unexpected internal server error occured while retrieving item taxes.");
+                return StatusCode((int)HttpStatusCode.InternalServerError, "Internal server error.");
+            }
```

Comment:
IMO you can just leave the `_itemService.GetItemTaxes(id);` and `return Ok(itemTaxes);` lines without try/catch block. Any generic Exception will be caught in our exception handler, logged, and then the Response to the client will be same as yours - Internal Server Error 

In logs we will able to see the stack trace, so it's not necessary to explicitly log "while retrieving item taxes" part

Reply:
Fair point, but to be fair, basically all of exception handling needs to be looked at/refactored (it is the same in other places), therefore I am not sure if it would be practical to do in this PR.